### PR TITLE
use scc flags during podman pull

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

use scc flags during podman pull

## Test coverage
- No tests

- [x] **DONE**

## Links

Issue(s): #

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by cbosdo — APPROVED** [review]
And now I already smell the question: "where are `--scc-*` parameters on `mgrpxy`?"... Same for kubernetes: we probably will need to add a secret to use for registry using those

**Comment by mbussolotto** [comment]
> And now I already smell the question: "where are --scc-* parameters on mgrpxy?

@cbosdo since it was a 5 min work, I added the flag also for mgrpxy :) I re-asked for review

File: shared/podman/login.go [hunk 1]
```diff
@@ -9,15 +9,27 @@ import (
...
-	if hostData.SccPassword != "" && hostData.SccUsername != "" {
+func PodmanLogin(hostData *HostInspectData, scc types.SCCCredentials) (string, func(), error) {
+	scc_user := hostData.SccUsername
+	scc_password := hostData.SccPassword
+	if scc.User != "" {
+		log.Info().Msg(L("scc-user flag will be used. SCCCredentials in host machine will be ignored."))
+		scc_user = scc.User
+	}
+	if scc.Password != "" {
+		log.Info().Msg(L("scc-password flag will be used. SCCCredentials in host machine will be ignored."))
```
Comment (cbosdo) [inline shared/podman/login.go:27]: I would group this warning with the other. If we have only `--scc-user` set we cannot pick the password from the host as it likely wouldn't match

File: shared/podman/login.go [hunk 2]
```diff
@@ -9,15 +9,27 @@ import (
...
 // PodmanLogin logs in the registry.suse.com registry if needed and returns an authentication file, a cleanup function and an error.
-func PodmanLogin(hostData *HostInspectData) (string, func(), error) {
-	if hostData.SccPassword != "" && hostData.SccUsername != "" {
+func PodmanLogin(hostData *HostInspectData, scc types.SCCCredentials) (string, func(), error) {
+	scc_user := hostData.SccUsername
+	scc_password := hostData.SccPassword
+	if scc.User != "" {
+		log.Info().Msg(L("scc-user flag will be used. SCCCredentials in host machine will be ignored."))
+		scc_user = scc.User
+	}
```
Comment (cbosdo) [inline shared/podman/login.go:25]:
```suggestion
	if scc.User != "" && scc.Password != "" {
		log.Info().Msg(L("SCC credentials parameters will be used. SCC credentials from host will be ignored."))
		scc_user = scc.User
		scc_password = scc.Password
	}
```

File: shared/utils/cmd.go [hunk 3]
```diff
@@ -54,6 +54,16 @@ func CommandHelper[T interface{}](
 	return fn(globalFlags, flags, cmd, args)
 }
 
+// AddSCCFlag add SCC flags to a command.
+func AddSCCFlag(cmd *cobra.Command) {
+	cmd.Flags().String("scc-user", "", L("SUSE Customer Center username"))
```
Comment (cbosdo) [inline shared/utils/cmd.go:59]: Now we would miss an important help here for `mgradm`: I believe the help should state that those credentials are used to pull the images if coming from registry.suse.com and that on `mgradm` they also set the SCC credentials in the configuration for the products synchronization.

I wonder where to add that piece of help though as it shouldn't be in `mgrpxy`

Comment (mbussolotto) [reply shared/utils/cmd.go:59]: probably better to create two different `AddSCCFlag` function, one for `mgradm` and one in `mgrpxy`, in two different files

**Pull request merged**